src/main/scala/is/hail/distributedmatrix/BlockMatrixIsDistributedMatrix.scala

```diff
@@ -171,9 +177,9 @@ object BlockMatrixIsDistributedMatrix extends DistributedMatrix[BlockMatrix] {
       new DenseMatrix(m.numRows, m.numCols, m.toArray.zipWithIndex.map { case (e, j) =>
         val rowsInThisBlock: Int = (if (blockRow + 1 == rowBlocks && rowsRemainder != 0) rowsRemainder else rowsPerBlock)
         val colsInThisBlock: Int = (if (blockCol + 1 == colBlocks && colsRemainder != 0) colsRemainder else colsPerBlock)
-        if (blockRow.toLong * rowsInThisBlock + j % rowsInThisBlock < nRows &&
-          blockCol.toLong * colsInThisBlock + j / rowsInThisBlock < nCols)
-          op(e, blockRow * rowsInThisBlock + j % rowsInThisBlock)
+        if (blockRow.toLong * rowsPerBlock + j % rowsInThisBlock < nRows &&
```

Comment:
Wait what, I definitely need the modulo. The array is a column-major matrix, to get the within-block row index, I need to mod by the number of rows stored in this block. Right?